Controller/PostController.php

```diff
@@ -107,38 +121,45 @@ public function collectionAction($collection)
             throw new NotFoundHttpException('Unable to find the collection');
         }
 
-        return $this->renderArchive(array('collection' => $collection), array('collection' => $collection));
+        return $this->renderArchive(array('collection' => $collection), array('collection' => $collection), $request);
     }
 
     /**
-     * @param string $year
-     * @param string $month
+     * @param string  $year
+     * @param string  $month
+     * @param Request $request
      *
      * @return Response
      */
-    public function archiveMonthlyAction($year, $month)
+    public function archiveMonthlyAction($year, $month, Request $request = null)
     {
+        $request = $this->resolveRequest($request);
+
         return $this->renderArchive(array(
             'date' => $this->getPostManager()->getPublicationDateQueryParts(sprintf('%d-%d-%d', $year, $month, 1), 'month'),
-        ));
+        ), $request);
     }
 
     /**
-     * @param string $year
+     * @param string  $year
+     * @param Request $request
      *
      * @return Response
      */
-    public function archiveYearlyAction($year)
+    public function archiveYearlyAction($year, Request $request = null)
     {
+        $request = $this->resolveRequest($request);
+
         return $this->renderArchive(array(
             'date' => $this->getPostManager()->getPublicationDateQueryParts(sprintf('%d-%d-%d', $year, 1, 1), 'year'),
-        ));
+        ), $request);
     }
 
     /**
      * @throws NotFoundHttpException
      *
      * @param $permalink
+     * @param Request $request
```

Comment:
you added a phpdoc, but you don't add ``$request`` to the signature of ``viewAction()``